File: plat/ts/entities/coin.ts
import * as SLT from 'slate2d';
import * as Draw from 'draw';
import * as Assets from 'assets';

import Entity, { Layer } from '../entity.js';
import Dir from '../dir.js';
import CollisionType from '../collisiontype.js';
import { World } from '../game.js';
import { Player } from './player.js';

class Coin extends Entity {
  type = 'Coin';
  collidable = CollisionType.Trigger;
  sprite = Assets.find('coin');
  layer = Layer.Background;

  constructor(args: { [key: string]: any }) {
    super(args);
  }

  update(ticks: number, dt: number) {
    this.frame = ticks / 8 % 4;
  }

  collide(other: Entity, dir: Dir) {
    if (other instanceof Player == false) {
      return;
    }

    this.destroyed = true;
    World().state.currCoins++;
  }
}

export { Coin };

File: csgame/entities/Coin.cs
using Slate2D;

[Spawnable]
class Coin : Entity {
  public Coin(LDTKEntity ent) : base(ent) {
    Collidable = CollisionType.Trigger;
    Sprite = Assets.Find("coin");
    Layer = Layer.Background;
  }

  public override void Collide(Entity other, Dir dir) {
    if (other is Player == false) return;

    Destroyed = true;
    Main.World.GameState.CurrentCoins++;
  }

  public override void Update(uint ticks, float dt) => Frame = ticks / 8 % 4;
}
